unumpy/_multimethods.py

```diff
@@ -1421,3 +1564,212 @@ def _vander_default(x, N=None, increasing=False):
+
+        A = concatenate(arrays, axis=axis)
+
+    return A
+
+
+@create_numpy(_first2argreplacer, default=_tile_default)
+@all_of_type(ndarray)
+def tile(A, reps):
+    return (A, reps)
+
+
+@create_numpy(_self_argreplacer)
+@all_of_type(ndarray)
+def repeat(a, repeats, axis=None):
+    return (a,)
+
+
+@create_numpy(_self_argreplacer)
+@all_of_type(ndarray)
+def delete(arr, obj, axis=None):
+    return (arr,)
+
+
+def _insert_argreplacer(args, kwargs, dispatchables):
+    def replacer(arr, obj, values, axis=None):
+        return (dispatchables[0], obj, dispatchables[1]), {"axis": axis}
+
+    return replacer(*args, **kwargs)
+
+
+@create_numpy(_insert_argreplacer)
+@all_of_type(ndarray)
+def insert(arr, obj, values, axis=None):
+    return (arr, values)
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def append(arr, values, axis=None):
+    return (arr, values)
+
+
+@create_numpy(_self_argreplacer)
+@all_of_type(ndarray)
+def resize(a, new_shape):
+    return (a,)
+
+
+def _trim_zeros_default(filt, trim="fb"):
+    nonzero_idxs = nonzero(filt)[0]
+    start, end = None, None
+    if "f" in trim:
+        start = nonzero_idxs[0]
+    if "b" in trim:
+        end = nonzero_idxs[-1] + 1
+    return filt[start:end]
+
+
+@create_numpy(_self_argreplacer, default=_trim_zeros_default)
+@all_of_type(ndarray)
+def trim_zeros(filt, trim="fb"):
+    return (filt,)
+
+
+def _flip_default(m, axis=None):
+    if axis is None:
+        axis = tuple(range(ndim(m)))
+    if isinstance(axis, numbers.Integral):
+        axis = (axis,)
+
+    axis = _normalize_axis(ndim(m), axis)
+
+    slices = []
+    for i in range(ndim(m)):
+        if i in axis:
+            slices.append(slice(None, None, -1))
+        else:
+            slices.append(slice(None, None, 1))
+
+    return m[tuple(slices)]
+
+
+@create_numpy(_self_argreplacer, default=_flip_default)
+@all_of_type(ndarray)
+def flip(m, axis=None):
+    return (m,)
+
+
+@create_numpy(_self_argreplacer, default=lambda m: flip(m, 1))
+@all_of_type(ndarray)
+def fliplr(m):
+    return (m,)
+
+
+@create_numpy(_self_argreplacer, default=lambda m: flip(m, 0))
+@all_of_type(ndarray)
+def flipud(m):
+    return (m,)
+
+
+def _to_normalized_split_index(dim, shift):
+    if shift >= 0:
+        index = dim - (shift % dim)
+    else:
+        index = abs(shift) % dim
+
+    return index
+
+
+def _roll_default(a, shift, axis=None):
+    if axis is None:
+        flatted = ravel(a)
+
+        index = _to_normalized_split_index(len(flatted), shift)
+        arrays = array_split(flatted, [index])
+
+        return concatenate(arrays[::-1]).reshape(a.shape)
+
+    if isinstance(axis, numbers.Integral):
+        axis = (axis,)
+    if isinstance(shift, numbers.Integral):
+        shift = (shift,)
+
+    axis = _normalize_axis(ndim(a), axis, allow_repeats=True)
+
+    if len(axis) >= len(shift):
+        shift = broadcast_to(shift, (len(axis),))
+    else:
+        axis = broadcast_to(axis, (len(shift),))
```

Comment:
Try to avoid using arrays to represent quantities along axes. Arrays are optimised for  large data sets, while number of axis tends to be small. This is especially true for CuPy where you pay a huge cost to transfer the data to the GPU.

Instead, I would conditionally use `itertools.repeat` to turn single axis into an Iterable of the same axis.